lib/expand/join.go

```diff
@@ -114,6 +112,11 @@ type Peer struct {
 	execDoneC chan install.ExecResult
 	// wg is a wait group used to ensure completion of internal processes
 	wg sync.WaitGroup
+	// connectC receives the results of connecting to either wizard
+	// or cluster controller
+	connectC chan connectResult
+	// connectingC is closed once the connect loop starts running
```

Comment:
I introduced this knob to handle the case with now delayed connect loop and possibility of agent using the operation context which is not available in this case. The operation context is required to know the operation details to be able to set the operation into failed state on exiting with error and with the operation not completed.
Since now, the service starts but does not automatically connect to either the installer or cluster controller unless explicitly requested by the client (effectively by one of  `./gravity join` or `./gravity resume`), the operation context is empty.
If you were to immediately stop the service in this state (i.e. with no connect loop running), it would get stuck waiting on the context to become available. To workaround this, I notify that the operation context should be available with this channel by closing it. While it is not closed, the service will not attempt to use the operation context.